src/main/java/io/tiledb/spark/Range.java

```diff
@@ -595,4 +595,63 @@ else if (leftOvers > 0) {
 
     return ranges;
   }
+
+  /**
+   * Split a range. Will split upto N buckets
+   *
+   * @param partitions The number of partitions
+   * @param partitionWidth The number of elements in the partition
+   * @return The list of ranges, one per partition
+   * @throws TileDBError
+   */
+  public List<Range> splitRangeToPartitions(Object partitions, Object partitionWidth)
+      throws TileDBError {
+    List<Range> ranges = new ArrayList<>();
+    // Number of buckets is 1 more thank number of splits (i.e. split 1 time into two buckets)
+    // Only long dimensions can be split with naive algorithm
+    Object min = range.getFirst();
+    Object max = range.getSecond();
+    Object currentMin;
+    Object currentMax;
+
+    // If the min is the max this range is not splittable
+    if (!this.splittable()) {
+      ranges.add(new Range(range));
+      return ranges;
+    }
+
+    Object one = null;
+    if (dataClassType == Byte.class) {
+      one = (byte) 1;
+    } else if (dataClassType == Short.class) {
+      one = (short) 1;
+    } else if (dataClassType == Integer.class) {
+      one = 1;
+    } else if (dataClassType == Long.class) {
+      one = 1L;
+    } else if (dataClassType == Float.class) {
+      one = 1f;
+    } else if (dataClassType == Double.class) {
+      one = 1d;
+    }
+
+    currentMin = min;
+    currentMax =
+        util.subtractObjects(
+            util.addObjects(currentMin, partitionWidth, dataClassType), one, dataClassType);
```

Comment:
Are you using `one` to set the max range to the `min+width-1`? If so for floats and doubles you need to use epsilon instead of `1f`/`1d`.

Reply:
I removed one completely and used `{add/subtract}Epsilon()` in all cases